### ISSUE TITLE
`Model` should inherit from `gym.Env`

### ISSUE THREAD

**Issue by eleurent** [rlberry-py/rlberry#18]
`gym.Env` is a well-established standard for environments. I think that our `Model` class should only be an addition, as it is basically the same with the added ability to define generative models, and the centralized seeding.
But it should not be an independent class as it is now, since it prevents using all the existing tools online tailored for `gym.Env`.

For instance, when I needed an episodic version of MountainCar to test DQN i did the following:

```
from rlberry.envs.classic_control import MountainCar
from gym.wrappers import TimeLimit

env = MountainCar()
env = TimeLimit(env, max_episode_steps=200)
```
which, results in an `AttributeError: 'MountainCar' object has no attribute 'metadata'`

The solution with the current mindset would be to define a custom version `rlberry.TimeLimit`, but I suspect that this approach will only end up in an entire replication of `gym.Env` to our definitions, which is not desirable.

TLDR: make `gym` the default interface, and inherit from it to include additional/modified features for specific usages
